akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala

```diff
@@ -773,37 +776,27 @@ object PartitionHub {
    *   the first is the number of active consumers and the second is the stream element. The function should
    *   return the index of the selected consumer for the given element, i.e. int greater than or equal to 0
    *   and less than number of consumers. E.g. `(size, elem) => math.abs(elem.hashCode) % size`.
-   * @param startAfterNbrOfConsumers Elements are buffered until this number of consumers have been connected.
+   * @param startAfterNrOfConsumers Elements are buffered until this number of consumers have been connected.
    *   This is only used initially when the stage is starting up, i.e. it is not honored when consumers have
    *   been removed (canceled).
    * @param bufferSize Total number of elements that can be buffered. If this buffer is full, the producer
    *   is backpressured.
    */
-  def sink[T](partitioner: (Int, T) ⇒ Int, startAfterNbrOfConsumers: Int,
+  def sink[T](partitioner: (Int, T) ⇒ Int, startAfterNrOfConsumers: Int,
               bufferSize: Int = defaultBufferSize): Sink[T, Source[T, NotUsed]] =
-    statefulSink(() ⇒ (info, elem) ⇒ info.consumerIds(partitioner(info.size, elem)), startAfterNbrOfConsumers, bufferSize)
+    statefulSink(() ⇒ (info, elem) ⇒ info.consumerIdByIdx(partitioner(info.size, elem)), startAfterNrOfConsumers, bufferSize)
 
   @DoNotInherit trait ConsumerInfo extends akka.stream.javadsl.PartitionHub.ConsumerInfo {
 
     /**
-     * Identifiers of current consumers
-     */
-    def consumerIds: Array[Long]
-
-    /**
-     * Approximate number of buffered elements for a consumer.
-     * Larger value than other consumers could be an indication of
-     * that the consumer is slow.
+     * Sequence of all identifiers of current consumers.
      *
-     * Note that this is a moving target since the elements are
-     * consumed concurrently.
+     * Use this method only if you need to enumerate consumer existing ids.
+     * When selecting a specific consumerId by its index, prefer using the dedicated [[consumerIdByIdx]] method instead,
+     * which is optimised for this use case.
      */
-    def queueSize(consumerId: Long): Int
+    def consumerIds: immutable.IndexedSeq[Long]
 
-    /**
-     * Number of attached consumers.
-     */
-    def size: Int
```

Comment:
I think it's better to have all methods in both scaladsl and javadsl even though it adds some duplication.
It makes it easier to see the full api, and users don't have to care about the implementation detail that we scaladsl extends javadsl here.